code/game/objects/items/weapons/rpd.dm

```diff
@@ -0,0 +1,303 @@
+/*Contains:
+	Rapid Pipe Dispenser
+*/
+
+/obj/item/weapon/rpd
+	name = "rapid pipe dispenser"
+	desc = "This device can rapidly dispense atmospherics and disposals piping, manipulate loose piping, and recycle any detached pipes it is applied to."
+	icon = 'icons/obj/tools.dmi' //Apparently we can't use double quotes here
+	icon_state = "rpd"
+	opacity = 0
+	density = 0
+	anchored = 0
+	flags = CONDUCT
+	force = 10
+	throwforce = 10
+	throw_speed = 3
+	throw_range = 5
+	w_class = 3
+	materials = list(MAT_METAL = 30000, MAT_GLASS = 5000)
+	origin_tech = "engineering=4;materials=2"
+	var/iconrotation = 0 //used to orient icons and pipes
+	var/initialised //This stops us unnecessarily checking the cache for icons
+	var/mode = 1 //Disposals, atmospherics
+	var/pipemenu = 1 //For nanoUI menus
+	var/pipetype = 1//For nanoUI menus
+	var/whatpipe = 0 //What kind of pipe is it? See code/game/machinery/pipe/construction.dm for a list of defines
+	var/whatdpipe = 0 //What kind of disposals pipe is it? See code/game/machinery/pipe/pipe_dispenser.dm for a list of defines
+	var/cooldown = 0 //Waiting for something to complete?
+	var/spawndelay = 4 //Adding a *slight* delay to the RPD, shouldn't be very noticeable
+	var/walldelay = 40 //How long should drilling into a wall take?
+	var/datum/effect/system/spark_spread/spark_system
+
+/obj/item/weapon/rpd/New()
+	spark_system = new /datum/effect/system/spark_spread()
+	spark_system.set_up(1, 0, src)
+	spark_system.attach(src)
+
+//List of things follow here
+
+//Pipename, pipe type (see construction.dm), pipe category, number of unique orientations, name of pipe icon, is this pipe bendy?
+var/list/pipelist = list(
```

Comment:
Sounds like you were doing something wrong then.

And even if you REALLY need keys, dear god use some more descriptive names.